core/src/main/java/io/onetable/delta/DeltaValueConverter.java

```diff
@@ -65,6 +65,7 @@ public static Object convertFromDeltaColumnStatValue(Object value, OneSchema fie
     } catch (DateTimeParseException parseException) {
       // fall back to parsing without offset
       DateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT_STR);
+      dateFormat.setLenient(false);
       dateFormat.setTimeZone(TIME_ZONE);
```

Comment:
Can you pull this into a method for getting a DateFormat so we have less places to update in the future?